rotkehlchen/exchanges/htx.py

```diff
@@ -298,17 +305,19 @@ def _query_deposits_withdrawals(
 
             try:
                 coin = asset_from_htx(movement['currency'])
-                movements.append(AssetMovement(
-                    timestamp=timestamp,
+                movements.extend(create_asset_movement_with_fee(
+                    timestamp=ts_sec_to_ms(timestamp),
                     location=Location.HTX,
-                    category=category,
-                    address=movement['address'],
-                    transaction_id=movement['tx-hash'],
+                    event_type=event_type,
                     asset=coin,
                     amount=deserialize_asset_amount(movement['amount']),
                     fee_asset=coin,
                     fee=deserialize_fee(movement.get('fee', '0')),
-                    link=str(movement['id']),
+                    unique_id=str(movement['id']),
+                    extra_data=maybe_set_transaction_extra_data(
+                        address=movement['address'],
+                        transaction_id=movement['tx-hash'],
```

Comment:
can't you combine this with the `id` and use that as the `unique_id`?

Reply:
unique_id is used in combination with the location to create the event_identitfier, so it simply needs to be unique on this exchange. I think this `id` should be unique on this exchange, so I'm not sure there's really any point in combining it with `tx-hash`, expecially since `tx-hash` may not even be present some of the time as mentioned in my reply to your comment about `get_key_if_has_val`.